AcademicManagerV1/src/com/academic/db/dao/CourseDAO.java

```diff
@@ -30,13 +32,17 @@ public CourseDAO(Connection conn) throws SQLException {
 				ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);
 		countStatement = dbConnection.prepareStatement("SELECT COUNT(*) FROM course WHERE isDeleted=0;",
 		ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);
+		
+		teachingStatement = dbConnection.prepareStatement(
+				"SELECT courseId,person.surname from teaching inner join person on teaching.personId = person.personId",
+				ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);
 
 		addStatement = dbConnection.prepareStatement(
-				"INSERT INTO course (courseId, title, cost, description, startingDate, endingDate, isActive, isDeleted) VALUES ('?','?','?','?','?','?','?','?')");
+				"INSERT INTO course (courseId, title, cost, description, startingDate, endingDate, isActive, isDeleted) VALUES ('null','?','?','?','?','?','?','0');");
```

Comment:
CourseId should be AUTO_INCREMENT, and you must not give it an explicit value here.